js/render/vertex_array_object.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+var assert = require('assert');
+
+module.exports = VertexArrayObject;
+
+function VertexArrayObject() {
+    this.boundProgram = null;
+    this.boundVertexBuffer = null;
+    this.boundVertexBuffer2 = null;
+    this.boundVertexOffset = null;
+    this.boundElementBuffer = null;
+    this.vao = null;
+}
+
+var reported = false;
+
+VertexArrayObject.prototype.bind = function(gl, program, vertexBuffer, vertexBuffer2, vertexOffset, elementBuffer) {
```

Comment:
I'll add a switch from `secondElement` to `element2` in https://github.com/mapbox/mapbox-gl-js/pull/2488/files